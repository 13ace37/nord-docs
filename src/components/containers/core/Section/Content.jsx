/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import CoreContent from "containers/core/Content";

/**
 * A container for content of the `Section` component.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const Content = styled(CoreContent)`
  padding-top: ${({ compact }) => !compact && "5em"};
  padding-bottom: ${({ compact }) => !compact && "5em"};
`;

export default Content;
