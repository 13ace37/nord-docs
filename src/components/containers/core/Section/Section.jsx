/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import PropTypes from "prop-types";
import styled from "styled-components";

import { motion } from "styles/theme";

import {
  baseBackgroundColor,
  primaryBackgroundColor,
  secondaryBackgroundColor,
  tertiaryBackgroundColor
} from "../shared/styles";

const variants = {
  base: baseBackgroundColor,
  primary: primaryBackgroundColor,
  secondary: secondaryBackgroundColor,
  tertiary: tertiaryBackgroundColor
};

/**
 * A base HTML component that represents a `<section>` with multiple base style variants.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section
 * @see https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_HTML_sections_and_outlines
 * @since 0.3.0
 */
const Section = styled.section`
  background-color: ${({ variant }) => variants[variant]};
  transition: background-color ${motion.speed.duration.transition.base.themeModeSwitch}ms ease-in-out;
  position: ${({ decorated }) => decorated && "relative"};
`;

Section.propTypes = {
  variant: PropTypes.oneOf(Object.keys(variants))
};

Section.defaultProps = {
  variant: "base"
};

export default Section;
