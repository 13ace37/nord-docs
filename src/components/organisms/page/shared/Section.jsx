/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import CoreSection from "containers/core/Section";

/**
 * A main container that applies section specific styles and CSS workarounds.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.6.0
 */
const Section = styled(CoreSection)`
  /* The relative position is necessary for sections that render absolute positioned decorations. */
  position: relative;
  /*
   * Due to a bug in Firefox there is a one pixel gap between the SVG section divider (skewed wave) of the previous
   * section and this section. The workaround moves this section up by the amount of pixels specified by the "gapWidth"
   * prop to ensure the gap is closed. Values larger than one pixel are necessary because the previous section is
   * affected by the same problem and applied the same workaround which results in a gap larger than one pixel between
   * these sections.
   */
  top: ${({ gapWidth }) => gapWidth && `-${gapWidth}px`};
  /* Some sections need to use special background colors to differ from the previous and/or next section. */
  background-color: ${({ bgColor }) => bgColor};
`;

export default Section;
