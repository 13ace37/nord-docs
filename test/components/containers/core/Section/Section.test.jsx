/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React, { Fragment } from "react";

import Section from "containers/core/Section";
import { renderWithTheme } from "nord-docs-test-utils";

describe("theme styles", () => {
  test("matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Fragment>
        <Section>Section Base Background</Section>
        <Section variant="primary">Section Primary Background</Section>
        <Section variant="secondary">Section Secondary Background</Section>
      </Fragment>
    );
    expect(container).toMatchSnapshot();
  });
});
