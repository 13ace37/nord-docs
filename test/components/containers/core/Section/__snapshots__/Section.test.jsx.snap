// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme styles matches the snapshot 1`] = `
.c0 {
  position: relative;
  background-color: #fff;
  -webkit-transition: background-color 400ms ease-in-out;
  transition: background-color 400ms ease-in-out;
}

.c1 {
  position: relative;
  background-color: #fbfbfc;
  -webkit-transition: background-color 400ms ease-in-out;
  transition: background-color 400ms ease-in-out;
}

.c2 {
  position: relative;
  background-color: #f8f9fb;
  -webkit-transition: background-color 400ms ease-in-out;
  transition: background-color 400ms ease-in-out;
}

<div>
  <section
    class="c0"
  >
    Section Base Background
  </section>
  <section
    class="c1"
  >
    Section Primary Background
  </section>
  <section
    class="c2"
  >
    Section Secondary Background
  </section>
</div>
`;
